### DISCUSSION TITLE
Do we need focus distance in TCA (and distortion) correction?

### DISCUSSION THREAD

**Discussion by uchrisu** [lensfun/lensfun#1640] [General]
When I investigated the embedded correction data in Sony raw files I found that this data depends on the focus distance. 

For vignetting the lensfun profile already has the option for various focus distances but we do not have this option for TCA and distortion.

For distortion the dependence on focus distance is rather small, although it is existent, but for TCA there are extreme changes when varying the focus distance. For example the blue shift on my Sigma 35mm 1.2 Art in the corner changes from about +1 pixel (on an 24MP sensor) at infinity to -4 pixels at the nearest focus distance. So without taking the distance into account, a TCA correction seems meaningless here. 

Would you agree that we should add the focal distance as option in TCA and distortion?

**Comment by Sturmflut** [comment]
I always suspected this. I guess it might become worse since manufacturers have to cut down on cost, and many are leaving it to software and those who edit the images later instead of improving the optics. Adobe doesn't even include TCA data in their lens converter profiles anymore since it can "easily be fixed in Photoshop".

Focal length should be an optional tag for TCA and distortion, yes. I see two problems, though:

- We could only ship support for that with a new stable release, which isn't on the horizon.
- The script that @bronger uses to convert the current database to a version 1 database might choke on it. If that happens, all the people out there still using the last stable version 0.3.2 on their systems won't get any updates anymore.

**Reply by tomato42** [reply]
Having scripts to create two versions of the databse, when one is simply the other, but with few fields omitted, doesn't sound like a complex thing to me... I'm pretty sure I could create a python script to do that in just few hours
